xwiki-commons-core/xwiki-commons-environment/xwiki-commons-environment-standard/src/main/java/org/xwiki/environment/System.java

```diff
@@ -130,7 +130,7 @@ public static ComponentManager initialize(File permanentDirectory, File resource
     {
         // Step 1: Initialize Component system
         EmbeddableComponentManager ecm = new EmbeddableComponentManager();
-        ecm.initialize(classLoader == null ? System.class.getClassLoader() : classLoader);
+        ecm.initialize(classLoader == null ? Thread.currentThread().getContextClassLoader() : classLoader);
```

Comment:
I don't think this is correct. This is for the standard environment, i.e. to be used outside of JavaEE.

Reply:
I've checked more and indeed if the thread context CL is not set, it'll fall back to the system CL so the code above should have the same effect as above, even though it's purely unnecessary in a non JavaEE environment IMO. Also it might be better to use the current CL instead of the system CL.

Reply:
At the very minimum this line requires a comment to explain the usage of the current thread CL.

Reply:
Some reference: https://www.javaworld.com/article/2077344/core-java/find-a-way-out-of-the-classloader-maze.html